### PR TITLE
adding cluster aware statistics to gp_toolkit

### PR DESCRIPTION
**Pull request by skahler-pivotal**
cluster aware version of postgres stats added to gp_toolkit

### CODE REVIEW

**Comment by skahler-pivotal** [comment]
@kainwen this PR still seems to be failing the test when it hits the resource queue check

**Comment by xinzweb** [comment]
@skahler-pivotal any update on this PR? @kainwen? 

This is monitored by https://greenplum.slack.com/messages/C9NHTUWUB. Please make progress on this one, or close it if it's done. Thanks.

File: src/backend/catalog/gp_toolkit.sql [hunk 1]
```diff
@@ -2193,6 +2193,626 @@ ORDER BY segid;
...
+--
+-- @doc:
+--        Record type to add segment id on to pg_stat_database results
+--
+--------------------------------------------------------------------------------
+
+CREATE TYPE gp_toolkit.gp_stat_database_t
+AS
+(
+  dbseg int,
```
Comment (slari-pivotal) [inline src/backend/catalog/gp_toolkit.sql]: How about we rename this column to `gp_segment_id` as this is how the segment id is exposed for user tables?

Same comment applies to other tables.

**Comment by xinzweb** [comment]
Marked `ToDo` label, since we do get traction on this PR. Thanks @doty-pivotal  and @slari-pivotal

**Comment by skahler-pivotal** [comment]
I was seeing this too. I'm not sure why but this seems to have triggered a check to fail further down in the regress testing. I opened this issue https://github.com/greenplum-db/gpdb/issues/4582

**Pull request closed without merging**